### ISSUE TITLE
Twoslash will be incorrectly run on certain scenarios

### ISSUE THREAD

**Issue by frencojobs** [shikijs/twoslash#77]
<p align="center">
<img src="https://user-images.githubusercontent.com/17830687/125190297-668b2700-e262-11eb-8df6-08e56b5846f7.png"/>
</p>

This part is problematic because it will assume ` ```ts title="twoslash"` as twoslash code. Just leaving it up this issue as a reminder, this will be fixed together with the `parseCodeFenceInfo` refactor.

**Comment by frencojobs** [comment]
I wrote a much more expressive parser from scratch with properly defined grammar rules plus tests and permutations of tests. Then I felt it was a bit out of scope with twoslash and could be reused in other libraries as well. So I separated it as a different project and published it out as [fenceparser](https://github.com/frencojobs/fenceparser). Unlike the current `parseCodeFenceInfo`, it returns a `Map` so there's a few APIs (not public ones) that needs to be changed. If I have time today, I'll use it and refactor twoslash with it. Or maybe tomorrow.
